File: serwer/src/main/java/pl/nakicompany/ewzwierzat/domain/Adopcja.java
package pl.nakicompany.ewzwierzat.domain;

import lombok.Data;

import javax.persistence.*;

@Entity
@Data
@Table(name="adopcja")
public class Adopcja {

    @Id
    @Column(name="id")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @OneToOne(optional=false, cascade = CascadeType.REMOVE)
    @JoinColumn(name="osoba_adoptujaca_id")
    private OsobaAdoptujaca osobaAdoptujaca;

    @OneToOne(optional=false)
    @JoinColumn(name="zwierze_id")
    private Zwierze zwierze;

}

File: web/src/app/model/adopcja.ts
import {OsobaAdoptujaca} from './osobaAdoptujaca';
import {Zwierze} from './zwierze';

export class Adopcja {
  id: number;
  osobaAdoptujacaDTO: OsobaAdoptujaca;
  zwierzeDTO: Zwierze;
  constructor() {
    this.id = 1;
  }
}
